storage/aptosdb/src/lib.rs

```diff
@@ -2070,6 +2069,47 @@ impl DbWriter for AptosDB {
     }
 }
 
+fn delete_state_kv_and_merkle_nodes_at_genesis_after_fast_sync(
+    state_kv_db: &DB,
+    state_merkle_db: Arc<DB>,
+    fast_sync_version: Version,
+    state_kv_batch: &SchemaBatch,
+    state_merkle_batch: &SchemaBatch,
+) -> Result<()> {
+    let tree_reader = StateMerkleDb::new(state_merkle_db.clone(), 0);
+    let mut iter = state_merkle_db.iter::<JellyfishMerkleNodeSchema>(ReadOptions::default())?;
+    iter.seek(&NodeKey::new_empty_path(0))?;
+    for item in iter {
+        let (node_key, node) = item?;
+
+        if node_key.version() != 0 {
+            break;
+        }
+
+        if let Node::Leaf(leaf_node) = node {
+            let state_key = leaf_node.value_index().0.clone();
+            let (version, _) = StateDb::get_state_value_with_version_by_version_ext(
+                state_kv_db,
+                &state_key,
+                fast_sync_version,
+            )?
+            .expect("Must exist.");
+            if version != 0
```

Comment:
would be great to add some documentation on the reasoning of having this condition?
My understanding is:
version != 0 seems to indicate the node value has been updated after fast_sync version
version == 0 means the value hasn't been updated. 
why do we only want to delete the nodes in one scenario?